functions/Invoke-DbaDiagnosticQuery.ps1

```diff
@@ -189,7 +241,14 @@ Then it will export the results to Export-DbaDiagnosticQuery.
             }
 
             if (!$instanceOnly) {
-                $databases = $server.Query("Select Name from sys.databases where name not in ('master', 'model', 'msdb', 'tempdb')")
+                if (-not $DatabaseName) {
+                    $databases = Get-DbaDatabase -SqlInstance $SqlInstance -ExcludeAllSystemDb  | select-object Name
```

Comment:
if you just need names, use (get-foo).Name

Reply:
also, reuse $instance instead of $SqlInstance (will be faster, reuse properly opened connection and doesn't need -sqlcredentials to be passed after the first connection attempt). Same thing on line 249